### PR TITLE
feat(probe): ability to configure `scheme`

### LINKED ISSUE #985: Add support for configuring scheme in http probes
Here is the k8s documentation https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#http-probes

### PR DESCRIPTION
**Pull request by rvshettigar**
Signed-off-by: rakshith <rakshith@yahooinc.com>

Fixes #985

### CODE REVIEW

**Comment by rvshettigar** [comment]
@iliapolo Can you please take a look at this PR when you get a chance?

File: src/probe.ts [hunk 1]
```diff
@@ -56,6 +56,18 @@ export interface ProbeOptions {
   readonly timeoutSeconds?: Duration;
 }
 
+export enum HttpGetScheme {
```
Comment (iliapolo) [inline src/probe.ts]:
```suggestion
export enum ConnectionScheme {
```

File: src/probe.ts [hunk 2]
```diff
@@ -66,6 +78,13 @@ export interface HttpGetProbeOptions extends ProbeOptions {
    * @default - defaults to `container.port`.
    */
   readonly port?: number;
+
+  /**
+   *  Scheme to use for connecting to the host (HTTP or HTTPS).
+   *
+   *  @default - Defaults to HTTP.
```
Comment (iliapolo) [inline src/probe.ts]: Where possible, and if they sufficiently convey meaning, we use concrete values to describe defaults.
 
```suggestion
   *  @default HttpGetScheme.HTTP.
```

File: src/_action.ts [hunk 3]
```diff
@@ -20,10 +21,11 @@ export class Action {
     return { command };
   }
 
-  public static fromHttpGet(container: Container, path: string, options: { port?: number } = { }): k8s.HttpGetAction {
+  public static fromHttpGet(container: Container, path: string, options: { port?: number; scheme?: HttpGetScheme } = { }): k8s.HttpGetAction {
     return {
       path,
       port: k8s.IntOrString.fromNumber(options.port ?? container.port ?? 80),
+      scheme: options.scheme,
```
Comment (iliapolo) [inline src/_action.ts]: We try surfacing k8s defaults as much as possible. It makes the manifest more clear, without having to wonder "what is the k8s default for this property". 

```suggestion
      scheme: options.scheme ?? HttpGetScheme.HTTP,
```

**Review by iliapolo — CHANGES_REQUESTED** [review]
@rvshettigar Thanks for this! 

Just a few minor fixups and we're good to merge

**Comment by iliapolo** [comment]
@rvshettigar Should I give this PR another round of review?

**Comment by rvshettigar** [comment]
@iliapolo  Sorry, I missed your comment. Yes please, the PR is ready for your review. Thanks.

**Review by iliapolo — APPROVED** [review]

**Pull request merged**